### PR TITLE
Reg 68/style progress tracker

### PR DESCRIPTION
**Pull request by robbie-sundstrom**
Here's an image of the finished Progress Tracker:

image

Here's what it looks like when you expand a certain course. Note that this could look much better, but I'm moving on to accomplish other more important tasks before code freeze. 

image

### CODE REVIEW

**Comment by arianaolson419** [comment]
Looks awesome! Did you push your resolve merge conflict commits?

**Comment by robbie-sundstrom** [comment]
Merge conflicts have been resolved and merged!

**Review by arianaolson419 — APPROVED** [review]

File: public/stylesheets/progress-tracker.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  font-size : 10pt;
+}
+
+.progressRatio {
+  font-size : 10pt;
+}
+
+.progressTypeHeader {
+  font-size : 14pt;
+}
```
Comment (logandavis) [inline public/stylesheets/progress-tracker.css:17]: nit: newline at end of file?

File: public/stylesheets/progress-tracker.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.trackerComponent {
+  height: 500px;
+
```
Comment (logandavis) [inline public/stylesheets/progress-tracker.css:3]: nit: no newline here, to match other fields

File: src/containers/ProgressTrackerContainer.jsx [hunk 3]
```diff
@@ -56,56 +57,72 @@ class ProgressTrackerContainer extends Component {
...
+      engrCourses,
+      sciCourses,
+      ahseCourses,
+    } = this.props;
+
     if (isGenReq) {
       const genreqItems = genreqs.map(req => <li>{req}</li>);
       return (
-        <div>
+        <div className={styles.trackerComponent}>
```
Comment (logandavis) [inline src/containers/ProgressTrackerContainer.jsx:85]: is this syntax a thing? does that import the classname from the CSS file? huh

File: src/containers/ProgressTrackerContainer.jsx [hunk 4]
```diff
@@ -140,6 +162,11 @@ ProgressTrackerContainer.propTypes = {
   engrTotal: PropTypes.number.isRequired,
   sciTotal: PropTypes.number.isRequired,
   ahseTotal: PropTypes.number.isRequired,
+  mathSciPerc: PropTypes.string.isRequired,
```
Comment (logandavis) [inline src/containers/ProgressTrackerContainer.jsx]: "Perc"onally, I am not a fan of this naming scheme. pct is a better abbreviation for percent, and I like whole words written up. But it seems like it'd be a real pain in the ass to change, so we can probably leave it for now.

File: src/reducers/ProgressTrackerReducer.jsx [hunk 5]
```diff
@@ -49,6 +49,20 @@ const ProgressTrackerReducer = (state = initialState, action) => {
       const ahseTotal = sum(ahseCourses);
       const sciTotal = sum(sciCourses);
 
+      // converts progress into a string that can be used by
+      // the progress bar in the ProgressTracker component
+      const mathSciPercNum = Math.round((mathTotal + sciTotal) * 3.33);
```
Comment (logandavis) [inline src/reducers/ProgressTrackerReducer.jsx:54]: This (hard-coded factors, toString() with concat) is super gross. Can this be a function or something? Or can we perhaps do the string conversion in the component's curly braces?

File: src/reducers/ProgressTrackerReducer.jsx [hunk 6]
```diff
@@ -49,6 +49,20 @@ const ProgressTrackerReducer = (state = initialState, action) => {
...
+      const engrPercNum = Math.round(engrTotal * 2.17);
+      const engrPerc = engrPercNum.toString().concat('%');
+      const ahsePercNum = Math.round(ahseTotal * 3.57);
+      const ahsePerc = ahsePercNum.toString().concat('%');
+      const genReqsPercNum = Math.round(genreqs.length * 6.25);
+      const genReqsPerc = genReqsPercNum.toString().concat('%');
+      const majorReqsPercNum = Math.round(majorreqs.length * 14.28);
+      const majorReqsPerc = majorReqsPercNum.toString().concat('%');
+
+
```
Comment (logandavis) [inline src/reducers/ProgressTrackerReducer.jsx:65]: nit: unnecessary blank line

**Review by logandavis — CHANGES_REQUESTED** [review]

File: src/reducers/ProgressTrackerReducer.jsx (hunk 5 above)
Comment (robbie-sundstrom) [reply src/reducers/ProgressTrackerReducer.jsx:54]: It wasn't working in the component's curly braces and I couldn't figure out why. In the interest of time, I just left it similar to this.

File: src/containers/ProgressTrackerContainer.jsx (hunk 3 above)
Comment (robbie-sundstrom) [reply src/containers/ProgressTrackerContainer.jsx:85]: Yeah this is how you get class names using css modules.

File: src/containers/ProgressTrackerContainer.jsx [hunk 7]
```diff
@@ -56,56 +57,72 @@ class ProgressTrackerContainer extends Component {
...
     } else if (isMajorReq) {
       const majorreqItems = majorreqs.map(req => <li>{req}</li>);
       return (
-        <div>
+        <div className={styles.trackerComponent}>
           <a onClick={this.backReq} href=""> ← back</a>
-          <p>{majorreqs.length}/7 courses</p>
-          <ul>{majorreqItems}</ul>
+          <p className={styles.progressRatio}>{majorreqs.length}/7 courses</p>
+          <ul className={styles.classList}>{majorreqItems}</ul>
```
Comment (yunhsincynthiachen) [inline src/containers/ProgressTrackerContainer.jsx]: Can we have some way of knowing that the courses listed below the back link are the completed ones?

**Review by logandavis — APPROVED** [review]

**Comment by yunhsincynthiachen** [comment]
awesome! looks good

**Pull request merged**